controllers/backstage_route.go

```diff
@@ -49,10 +49,12 @@ func (r *BackstageReconciler) reconcileBackstageRoute(ctx context.Context, backs
 		if errors.IsConflict(err) {
 			return fmt.Errorf("retry sync needed: %v", err)
 		}
-		setStatusCondition(backstage, bs.RouteSynced, metav1.ConditionFalse, bs.SyncFailed, fmt.Sprintf("Error:%s", err.Error()))
-		return err
+		msg := fmt.Sprintf("failed to sync Backstage Route: %s", err.Error())
+		setStatusCondition(backstage, bs.ConditionSynced, metav1.ConditionFalse, bs.SyncFailed, msg)
+		setStatusCondition(backstage, bs.ConditionRouteSynced, metav1.ConditionFalse, bs.SyncFailed, msg)
+		return fmt.Errorf(msg)
 	}
-	setStatusCondition(backstage, bs.RouteSynced, metav1.ConditionTrue, bs.SyncOK, fmt.Sprintf("Route host:%s", route.Spec.Host))
+	setStatusCondition(backstage, bs.ConditionRouteSynced, metav1.ConditionTrue, bs.SyncOK, fmt.Sprintf("Route host: %s", route.Spec.Host))
```

Comment:
Can not we have single "running" condition for backstage and contribute there the state and the message?
Indeed, if Backstage pod is running well but you can not route it, does it mean Backstage is working as expected?

Reply:
The sync status and runtime object status must be handled separately. If we only use one condition, it would be complicated to handle the situation such as: 1) the backstage CR is deployed/synced and and backstage app is up and running, and then 2) the CR is updated with an incorrect configuration and  sync would fail, but the backstage application is still up and running (since the CR change is not applied).

Reply:
yes, it is two different statuses (we've had it) but in my definitions:
- synced means CR state is synced with runtime 
- running means runtime is functional

Reply:
> yes, it is two different statuses (we've had it) but in my definitions:
> 
> * synced means CR state is synced with runtime
> * running means runtime is functional
Yeah this sounds good to me.  As you suggested, we can wait for feedback from customers and see how  "Running" status should be implemented.